### DISCUSSION TITLE
generate srt file with python

### DISCUSSION THREAD

**Discussion by evafans** [deepgram/community#486] [General help] [answered]
I‘m not a coder, I uesd chatgpt to help me write a python script and it never worked. I need to transcribe a local file using python, your tutorials doesn't seem helpful, how should I edit my script?

import deepgram

api_key = "YOUR_API_KEY"
api_secret = "YOUR_API_SECRET"

client = deepgram.Deepgram(api_key, api_secret)

audio_path = "/path/to/audio.wav"

transcript = client.transcribe(audio_path, format="srt")

with open("output.srt", "w") as f:
    f.write(transcript)

error:
AttributeError: module 'deepgram' has no attribute 'Client'

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @evafans, you can use our Python SDK to get an SRT file:

```
from deepgram import Deepgram
deepgram = Deepgram("<your API key>")
response = deepgram.transcription.sync_prerecorded({"url": "https://static.deepgram.com/examples/Bueller-Life-moves-pretty-fast.wav"}, {'punctuate': True, 'smart_format': True, 'model': 'nova', 'utterances': True})
print(deepgram.extra.to_SRT(response))
```

```
1
00:00:05,519 --> 00:00:06,019
Yep.

2
00:00:07,094 --> 00:00:08,615
I said it before, and I'll say it

3
00:00:08,615 --> 00:00:09,115
again.
```

Docs: [Automatically Generating WebVTT and SRT Captions](https://developers.deepgram.com/docs/automatically-generating-webvtt-and-srt-captions)

**Comment by evafans** [comment]
what if I want to transcribe a local file?

**Comment by evafans** [comment]
any updates?

**Comment by jjmaldonis** [comment]
Hey @evafans here's an example from our docs showing how to upload a local file: https://developers.deepgram.com/docs/getting-started-with-pre-recorded-audio#write-the-code

You can also check our Python SDK documentation. Our Python SDK is about to be released with a major version upgrade, so make sure to look at the code for the [specific release version](https://github.com/deepgram/deepgram-python-sdk/releases) you're using.

**Comment by jpvajda** [comment, marked as answer]
@evafans  Here is one more library to checkout. We recently released a [standalone package for Python to handle WebVTT and SRT formatting](https://github.com/deepgram/deepgram-python-captions) this can be used as a stand alone implementation without the SDK mentioned above. It just depends on what approach you want to take. Using the stand alone package might be a simpler approach.
